packages/core/src/TextSplitter.ts

```diff
@@ -55,10 +53,9 @@ export function cjkSentenceTokenizer(sentence: string): string[] {
   return slist.filter((s) => s.length > 0);
 }
 
-export const defaultParagraphSeparator = EOL + EOL + EOL;
-
-// In theory there's also Mac style \r only, but it's pre-OSX and I don't think
-// many documents will use it.
+// NOTE this is different from Python because in Python a lot of the heavy lifting for paragraph separation
+// is already handled by NLTK, but we don't have that option here.
+export const defaultParagraphSeparator = /\n\n|\r\n\r\n|\r\r/g;
```

Comment:
This way we decouple the document encoding from the OS that the user is currently running on.